summarization/text_summarization.py

```diff
@@ -32,12 +44,41 @@ def __init__(self, priority):
         '''
         self.text_summarizer = None
         if priority == "accuracy":
-            self.text_summarizer = Summarizer(model='distilbert-base-uncased')
+            handler = CoreferenceHandler(greedyness=.4)
+            self.text_summarizer = Summarizer(
+                model='distilbert-base-uncased',
+                sentence_handler=handler)
         elif priority == "speed":
             self.text_summarizer = summarize
         else:
             raise IncorrectInputError(
-                    "priority must be either accuracy or speed")
+                "priority must be either accuracy or speed")
+
+    def _get_entites_from_text(self):
+        entity_retriever = TextEntityRetriever()
+        self.entity_list = entity_retriever.get_entities_from_text(self.text)
+
+    def _cleaned_and_word_tokenized(self, text):
+        # strip punctuations
+        text = text.translate(str.maketrans('', '', string.punctuation))
+        word_tokenized_text = word_tokenize(text)
+        return [
+            word.lower() for word in word_tokenized_text if word.isalnum()
+        ]
+
+    def _list_has_sublist(self, list_1, list_2):
+        '''
+        Returns true if list_1 has list_2
+        as a sublist(subarray)
+        '''
+        list_1_len = len(list_1)
+        list_2_len = len(list_2)
+        for i in range(list_1_len):
+            if i + list_2_len - 1 < list_1_len \
+                    and list_1[i:i + list_2_len] == list_2:
+                return True
```

Comment:
Is list_1_len > list2_len?
```suggestion
        for i in range(list_1_len - list2_len + 1):
            if list_1[i:i + list_2_len] == list_2:
                return True
```

Reply:
list_1_len need not be always > list_2_len.
Regardless i guess your implementation is cleaner and should work!